app/AcquireIntersonArrayRF.cxx

```diff
@@ -217,10 +182,18 @@ int main( int argc, char * argv[] )
   Sleep( 100 ); // "time to stop"
   container.DisposeScan();
 
-  typedef itk::ImageFileWriter< ImageType > WriterType;
+  //
+  typedef itk::Image<short> WriterImageType;
+  typedef itk::CastImageFilter<ImageType, WriterImageType> CasterType;
```

Comment:
Ok.  In IntersonArrayCxxImagingContainer, perhaps we need two different pixel types: 
RFPixelType and RFImagePixelType
The new RFImagePixelType will be short (correct for the data)
The NewRFImageHandler will have a NativeRFBuffer of RFImagePixelType and the HandleNewRFImage event handler will do the type conversion when copying the managed buffer into the native buffer - and it will pass the (short) nativebuffer to the C++ callback function.  See lines 49, 57-58, and 77-82 in src/IntersonArrayCxxImagingContainer.cxx